src/dune_rules/command.ml

```diff
@@ -95,7 +95,9 @@ and expand_no_targets ~dir (t : without_targets t) =
 
 let dep_prog = function
   | Ok p -> Action_builder.path p
-  | Error _ -> Action_builder.return ()
+  | Error not_found ->
+    Action_builder.fail
+      { fail = (fun () -> Action.Prog.Not_found.raise not_found) }
```

Comment:
I don't think this helps. `Error _` will already turn into the appropriate error at action execution (see Action_exec.ml)

Reply:
Sorry, I don't understand. Is the suggestion/request to return a `Result.t` value from this function? If so, I don't see how one can still make the call to `Action_builder.with_no_targets` below, which expects `unit Action_builder.t`?

To illustrate this code is doing something, I reverted the change in https://github.com/ocaml/dune/pull/6403/commits/2cbe15e0d48790f487c3d54e5ccd2d5726a30715. The tests show how the part of the error message that was explaining that `melc` is not available is gone.